mep/accounts/views.py

```diff
@@ -1,6 +1,77 @@
 from dal import autocomplete
 from django.db.models import Q
-from .models import Account
+from django.http import JsonResponse
+
+from mep.accounts.models import Account
+from mep.accounts.queryset import AddressSolrQuerySet
+from mep.people.views import MembersList
+
+
+class AddressList(MembersList):
+    '''JSON-only view. Returns addresses and members for display
+    map display; honors all the same filters and search options
+    as :class:`~mep.people.views.MembersList`.'''
+
+    # solr member name fields for keyword se
+    name_search_fields = ['name_t', 'sort_name_t', 'name_ngram']
+
+    def get_queryset(self):
+        # MembersList get_queryset returns a person solr queryset
+        # filtered according to query params in the request
+        members = super().get_queryset()
+        addresses = AddressSolrQuerySet().all()
+
+        # NOTE: handle search term specifically, since
+        # qf name query doesn't work with the join query.
+        # Search terms and raw query parameters on the members
+        # quereset are ignored.
+        search_term = self.request.GET.get('query', None)
+        if search_term:
+            # if there is a search term, join addresses on member slug to
+            # find address by member
+            # the member list view uses !qf notation, but that can't be
+            # combined with the join; convert to an OR search on all
+            # included fields, since we don't care about score here
+            name_search = ' OR '.join('%s:(%s)' % (field, search_term)
```

Comment:
Is there a performance impact to this? Don't think we have a choice, and I'm glad you found a solution, regardless!

Reply:
I didn't notice any performance issues with it. Glad I found a solution too! Took me a while.